src/Command/SitemapGeneratorCommand.php

```diff
@@ -15,21 +15,83 @@
 
 use Sonata\Exporter\Handler;
 use Sonata\Exporter\Writer\SitemapWriter;
-use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Sonata\SeoBundle\Sitemap\SourceManager;
+use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\Filesystem\Filesystem;
 use Symfony\Component\Finder\Finder;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RouterInterface;
 
 /**
  * Create a sitemap.
  *
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
  */
-class SitemapGeneratorCommand extends ContainerAwareCommand
+class SitemapGeneratorCommand extends Command implements ContainerAwareInterface
 {
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var SourceManager
+     */
+    private $sitemapManager;
+
+    /**
+     * @var Filesystem
+     */
+    private $fs;
+
+    /**
+     * @deprecated
+     *
+     * @var ContainerInterface|null
+     */
+    private $container;
+
+    public function __construct(?RouterInterface $router = null, ?SourceManager $sitemapManager = null, ?Filesystem $fs = null)
+    {
+        $this->router = $router;
+        $this->sitemapManager = $sitemapManager;
+        $this->fs = $fs;
+
+        parent::__construct();
+    }
+
+    /**
+     * @deprecated
+     *
+     * @todo Remove deprecated methods, remove interface implementation, cleanup 'use' block.
```

Comment:
Here is the marker we use for this:
```suggestion
     * NEXT_MAJOR Remove deprecated methods, remove interface implementation, cleanup 'use' block.
```